### ISSUE TITLE
class_cast_exception: class org.apache.lucene.util.BytesRef cannot be cast to class java.lang.String 

### ISSUE THREAD

**Issue by manuelsancheznec** [crate/crate#13751]
### CrateDB version

4.8.2

### Problem description

It seems that the problem is inside the ELASTIC SEARCH EMBEDDED driver, there is a broken index for that filter, when we go by ID it does not go by that index.

It could be similar to this discussion: https://discuss.elastic.co/t/java-lang-double-cannot-be-cast-to-org-apache-lucene-util-bytesref/27733

Error: NotSerializableExceptionWrapper[class_cast_exception: class org.apache.lucene.util.BytesRef cannot be cast to class java.lang.String (org.apache.lucene.util.BytesRef is in unnamed module of loader 'app'; java.lang.String is in module java.base of loader 'bootstrap')]

### Steps to Reproduce

- Record at position 499999, returns results
![image](https://user-images.githubusercontent.com/126877395/222715613-5d3886de-1ab2-4cad-a324-4563a6202079.png)

-Record that is in position 500000 does NOT show it
![image](https://user-images.githubusercontent.com/126877395/222715838-bcc9e33d-d99a-4385-98a4-fce1e9f07bb5.png)

- If we use the same OFFSET as in the first query, it does not retrieve records either.
![image](https://user-images.githubusercontent.com/126877395/222716020-22212e47-c52e-4068-9bce-862fda5b2870.png)

-However, if we filter by entity_ts greater than that of record 499999 , it shows us the following records.
![image](https://user-images.githubusercontent.com/126877395/222716174-33eee8d3-95ca-41f3-b134-4aa5d58bc3b6.png)

CREATE TABLE:
```sql
CREATE TABLE IF NOT EXISTS "doc"."raw_waze_jams" (
   "blockingalertuuid" TEXT,
   "city" TEXT,
   "country" TEXT,
   "delay" BIGINT,
   "entity_id" TEXT NOT NULL,
   "entity_location" GEO_POINT,
   "entity_ts" TIMESTAMP WITH TIME ZONE NOT NULL,
   "entity_type" TEXT NOT NULL,
   "entitytype" TEXT,
   "fiware_service" TEXT NOT NULL,
   "fiware_servicepath" TEXT NOT NULL,
   "length" BIGINT,
   "level" BIGINT,
   "line" ARRAY(OBJECT(DYNAMIC) AS (
      "x" REAL,
      "y" REAL
   )),
   "position" GEO_SHAPE INDEX USING GEOHASH,
   "roadtype" TEXT,
   "segments" ARRAY(OBJECT(DYNAMIC)),
   "speed" REAL,
   "speedkmh" REAL,
   "status" TEXT,
   "street" TEXT,
   "turntype" TEXT,
   "uuid" TEXT,
   "validity_ts" TIMESTAMP WITH TIME ZONE,
   "endnode" TEXT,
   "pubmillis" TEXT,
   PRIMARY KEY ("entity_id", "entity_ts", "entity_type", "fiware_service", "fiware_servicepath")
)
CLUSTERED INTO 6 SHARDS
WITH (
   "allocation.max_retries" = 5,
   "blocks.metadata" = false,
   "blocks.read" = false,
   "blocks.read_only" = false,
   "blocks.read_only_allow_delete" = false,
   "blocks.write" = false,
   codec = 'default',
   column_policy = 'dynamic',
   "mapping.total_fields.limit" = 1000,
   max_ngram_diff = 1,
   max_shingle_diff = 3,
   number_of_replicas = '1',
   "routing.allocation.enable" = 'all',
   "routing.allocation.total_shards_per_node" = -1,
   "store.type" = 'fs',
   "translog.durability" = 'REQUEST',
   "translog.flush_threshold_size" = 536870912,
   "translog.sync_interval" = 5000,
   "unassigned.node_left.delayed_timeout" = 60000,
   "write.wait_for_active_shards" = '1'
)
```

### Actual Result

`NotSerializableExceptionWrapper[class_cast_exception: class org.apache.lucene.util.BytesRef cannot be cast to class java.lang.String (org.apache.lucene.util.BytesRef is in unnamed module of loader 'app'; java.lang.String is in module java.base of loader 'bootstrap')]`

### Expected Result

fix embedded driver

**Comment by hlcianfagna** [comment]
Perhaps related with https://github.com/crate/crate/pull/13235 which was fixed in 5.1.1 ?

**Comment by manuelsancheznec** [comment]
> Perhaps related with #13235 which was fixed in 5.1.1 ?

Thank you so much for your reply. We will migrate our Cratedb 4.8.2 to 5.2.3

Best Regards.
